pkg/analysis/numericbounds/analyzer.go

```diff
@@ -0,0 +1,249 @@
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/utils"
+	"sigs.k8s.io/kube-api-linter/pkg/markers"
+)
+
+const (
+	name = "numericbounds"
+)
+
+// Type bounds for validation.
+const (
+	maxInt32   = 2147483647                                      // 2^31 - 1
+	minInt32   = -2147483648                                     // -2^31
+	maxFloat32 = 3.40282346638528859811704183484516925440e+38    // max float32
+	minFloat32 = -3.40282346638528859811704183484516925440e+38   // min float32
+	maxFloat64 = 1.797693134862315708145274237317043567981e+308  // max float64
+	minFloat64 = -1.797693134862315708145274237317043567981e+308 // min float64
+)
+
+// JavaScript safe integer bounds for int64 (±2^53-1).
+// Per Kubernetes API conventions, int64 fields should use bounds within this range
+// to ensure compatibility with JavaScript clients.
+const (
+	maxSafeInt64 = 9007199254740991  // 2^53 - 1 (max safe integer in JS)
+	minSafeInt64 = -9007199254740991 // -(2^53 - 1) (min safe integer in JS)
+)
+
+var errMarkerMissingValue = errors.New("marker value not found")
+
+// Analyzer is the analyzer for the numericbounds package.
+// It checks that numeric fields have appropriate bounds validation markers.
+var Analyzer = &analysis.Analyzer{
+	Name:     name,
+	Doc:      "Checks that numeric fields (int32, int64, float32, float64) have appropriate minimum and maximum bounds validation markers",
+	Run:      run,
+	Requires: []*analysis.Analyzer{inspector.Analyzer},
+}
+
+func run(pass *analysis.Pass) (any, error) {
+	inspect, ok := pass.ResultOf[inspector.Analyzer].(inspector.Inspector)
+	if !ok {
+		return nil, kalerrors.ErrCouldNotGetInspector
+	}
+
+	// Check fields in structs
+	inspect.InspectFields(func(field *ast.Field, _ extractjsontags.FieldTagInfo, markersAccess markershelper.Markers, qualifiedName string) {
+		// Create TypeChecker with closure capturing markersAccess and qualifiedName
+		typeChecker := utils.NewTypeChecker(isNumericType, func(pass *analysis.Pass, expr ast.Expr, node ast.Node, _ string) {
+			checkNumericTypeExpr(pass, expr, node, markersAccess, qualifiedName)
+		})
+
+		typeChecker.CheckNode(pass, field)
+	})
+
+	// Check type declarations (type aliases)
+	inspect.InspectTypeSpec(func(typeSpec *ast.TypeSpec, markersAccess markershelper.Markers) {
+		// Create TypeChecker with closure capturing markersAccess and type name
+		typeChecker := utils.NewTypeChecker(isNumericType, func(pass *analysis.Pass, expr ast.Expr, node ast.Node, _ string) {
+			checkNumericTypeExpr(pass, expr, node, markersAccess, typeSpec.Name.Name)
+		})
+
+		typeChecker.CheckNode(pass, typeSpec)
+	})
+
+	return nil, nil //nolint:nilnil
+}
+
+// isNumericType checks if the expression is a numeric type we want to validate.
+func isNumericType(pass *analysis.Pass, expr ast.Expr) bool {
+	ident, ok := expr.(*ast.Ident)
+	if !ok {
+		return false
+	}
+
+	// Only check int32, int64, float32, and float64 types
+	switch ident.Name {
+	case "int32", "int64", "float32", "float64":
```

Comment:
What about other types of integer? All the ones we typically forbid in the `integers` linter. We probably want to still report limits on those even though we advise against them generally